config/src/main/java/com/typesafe/config/ConfigFactory.java

```diff
@@ -1063,4 +1112,10 @@ private static ConfigLoadingStrategy getConfigLoadingStrategy() {
             return new DefaultConfigLoadingStrategy();
         }
     }
+
+    private static Boolean getOverrideWithEnv() {
+        String overrideWithEnv = System.getProperties().getProperty(OVERRIDE_WITH_ENV_PROPERTY_NAME);
+
+        return Boolean.parseBoolean(overrideWithEnv);
```

Comment:
if the value of the property is not a boolean this will throw an ugly and likely-mysterious exception right? might want to catch it and say something like "the value of OVERRIDE_WITH_ENV_PROPERTY_NAME must be a boolean not '%s'" 

Not sure if we already have this bug when looking at any other env var values

Reply:
I've deliberately choose to use [Boolean.parseBoolean](https://docs.oracle.com/javase/7/docs/api/java/lang/Boolean.html#parseBoolean(java.lang.String)) because it doesn't throw exceptions.
Basically it always return false unless the string is (case-insensitive) `true`.